api/src/main/java/com/netflix/iceberg/DataFile.java

```diff
@@ -127,6 +129,12 @@ static StructType getType(StructType partitionType) {
    */
   Map<Integer, ByteBuffer> upperBounds();
 
+  /**
+   * @return metadata about how this file is encrypted, or null if the file is stored in plain
+   *         text.
+   */
+  EncryptionKeyMetadata keyMetadata();
```

Comment:
In this case and elsewhere I like specifically denoting what the meaning of the returned byte buffer is. This allows code that uses the encryption managers to pass the returned key metadata directly into the key manager methods without wrapping it in `EncryptionKeyMetadata` first.

Reply:
Why not pass ByteBuffer to the key manager, too? Since EncryptionKeyMetadata is just a byte buffer, I don't think it makes much sense to wrap it in an object. That just makes it harder to read and write. For example: `return keyMetadata == null ? null : keyMetadata.keyMetadata();`

Reply:
Yup that sounds reasonable.